client/web/src/site-admin/AccessRequestsPage/index.tsx

```diff
@@ -0,0 +1,251 @@
+        APPROVE_ACCESS_REQUEST
+    )
+
+    const generateUsername = useGenerateUsername()
+
+    const handleApprove = useCallback(
+        (id: string, name: string, email: string): void => {
+            if (!confirm('Are you sure you want to approve the selected access request?')) {
+                return
+            }
+            eventLogger.log('AccessRequestApproved', { id })
+            async function createUserAndApproveRequest(): Promise<void> {
+                const username = await generateUsername(name)
+                const { data } = await createUser({
+                    variables: {
+                        email,
+                        username,
+                    },
+                })
+
+                if (!data) {
+                    throw new Error('No data returned from approveAccessRequest mutation')
+                }
+
+                await approveAccessRequest({
+                    variables: {
+                        id,
+                    },
+                })
+
+                setLastApprovedUser({
+                    username,
+                    email,
+                    resetPasswordURL: data?.createUser.resetPasswordURL,
+                })
+                await refetch()
+            }
+            createUserAndApproveRequest().catch(error => {
+                setError(error)
+                // eslint-disable-next-line no-console
+                console.error(error)
+            })
+        },
+        [generateUsername, createUser, approveAccessRequest, refetch]
+    )
+
+    return (
+        <>
+            <SiteAdminPageTitle icon={{ svgPath: mdiAccount, ariaLabel: 'Account requests icon' }}>
+                <span>Users</span>
+                <span>Account requests</span>
+            </SiteAdminPageTitle>
+            <Card className="p-3">
+                {[queryError, error].filter(Boolean).map((err, index) => (
+                    <Alert variant="danger" key={index}>
+                        {err?.message}
+                    </Alert>
+                ))}
+                {lastApprovedUser && (
+                    <AccountCreatedAlert
+                        email={lastApprovedUser.email}
+                        username={lastApprovedUser.username}
+                        resetPasswordURL={lastApprovedUser.resetPasswordURL}
+                    />
+                )}
+                <div className="d-flex justify-content-end">
+                    <PageSwitcher
+                        totalCount={connection?.totalCount ?? null}
+                        totalLabel={connection?.totalCount === 1 ? 'account request' : 'account requests'}
+                        {...paginationArgs}
+                    />
+                </div>
+                <Grid columnCount={5}>
+                    {['Email', 'Name', 'Last requested at', 'Extra Details', ''].map((value, index) => (
```

Comment:
`Last requested at` seems misleading, since it is showing `Created at` instead. Should we change it? Or change DB layer to update the date when the user tries to create an access request even if they did already beforehand. That would represent last requested at better.
```suggestion
                    {['Email', 'Name', 'Created at', 'Extra Details', ''].map((value, index) => (
```

Reply:
Thank you, updated!